src/hooks/useDeviceStateTracker.ts

```diff
@@ -0,0 +1,67 @@
+import type { Ref } from '@muban/muban';
+import { onMounted, onUnmounted, ref } from '@muban/muban';
+
+import DeviceStateTracker, { DeviceStateEvent } from 'seng-device-state-tracker';
+import type IDeviceStateData from 'seng-device-state-tracker/lib/IDeviceStateData';
+
+import sharedVariables from '../config/shared-variables.json';
+
+const cleanMediaQueries = Object.keys(sharedVariables.mediaQueries).reduce<{
+  [key: string]: string;
+}>((result, key: string) => {
+  // eslint-disable-next-line no-param-reassign
+  result[key] = sharedVariables.mediaQueries[key].replace(/'/g, '');
+  return result;
+}, {});
+
+let deviceStateTracker: DeviceStateTracker | null = null;
+
+export type DeviceState = IDeviceStateData['state'];
+export type DeviceStateName = IDeviceStateData['name'];
+
+/**
+ * This hook can be used to access the active device state
+ *
+ * Example:
+ * ```ts
+ * const { state, name } = useDeviceStateTracker();
+ * ```
+ */
+export const useDeviceStateTracker = (): {
+  state: Ref<DeviceState>;
+  name: Ref<DeviceStateName>;
+} => {
+  if (deviceStateTracker === null) {
+    deviceStateTracker = new DeviceStateTracker({
+      deviceState: sharedVariables.deviceState,
+      mediaQueries: cleanMediaQueries,
+      showStateIndicator: process.env.NODE_ENV === 'development',
+    });
+  }
+
+  const activeDeviceState = ref<IDeviceStateData['state']>(
+    deviceStateTracker.currentDeviceState.state,
+  );
+  const activeDeviceStateName = ref<IDeviceStateData['name']>(
```

Comment:
I would not recommend storing the name and the state in a different ref. They both belong to each other, so if 1 changes, the other should change too. So you better store them together in 1 ref

Reply:
I'll group them in one `ref` but in the return statement I would still prefer to have them separated because you usually only use the state and not the name and grouping them together would mean an extra layer every time you want to use the state value:

```ts
const deviceStateData  = useDeviceStateTracker();
if(deviceStateData.value.state > sharedVariables.deviceState.MIN_X)  ...

// versus
const { state }  = useDeviceStateTracker();
if(state.value > sharedVariables.deviceState.MIN_X)  ...
```